### PR TITLE
[Improvement-13146][Task-Relation] Support calling taskRelation api t…

### LINKED ISSUE #13146: [Improvement][Task-Relation] Support calling taskRelation api through projectCode and processCode to query taskCode
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/dolphinscheduler/issues?q=is%3Aissue) and found no similar feature requirement.

### Description

When modifying the workflow definition, if I do not know the taskcode, I will not be able to modify the workflow definition. Therefore, can DS support calling the api through projectCode and processCode to obtain the corresponding taskCode, so that I can easily modify it.

对于修改工作流定义时，假如我不知道taskcode，那么我将无法修改工作流定义，所以ds 能否支持通过projectCode和processCode调用api去获取相应的taskCode，这样我可以方便修改。

### Are you willing to submit a PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by LiJie20190102**
…hrough projectCode and processCode to query taskCode

## Purpose of the pull request

Support calling taskRelation api through projectCode and processCode to query taskCode

close #13146
## Verify this pull request

This pull request is code cleanup without any test coverage.

*(or)*

This change added tests and can be verified as follows:

Added org.apache.dolphinscheduler.api.controller.ProcessTaskRelationControllerTest#testQueryProcessTaskRelation to verify the change.

Added org.apache.dolphinscheduler.api.service.ProcessTaskRelationServiceTest#testQueryProcessTaskRelation to verify the change.

(or)

If your pull request contain incompatible change, you should also add it to `docs/docs/en/guide/upgrede/incompatible.md`

### CODE REVIEW

**Review by SbloodyS — COMMENTED** [review]
Could you please link this pr to an issue and describe why do you want to do that?

**Comment by LiJie20190102** [comment]
> Could you please link this pr to an issue and describe why do you want to do that?

The issue is [https://github.com/apache/dolphinscheduler/issues/13146](url). And the reason is when modifying the workflow definition, if I do not know the taskcode, I will not be able to modify the workflow definition. Therefore, can DS support calling the api through projectCode and processCode to obtain the corresponding taskCode, so that I can easily modify it.

**Comment by SbloodyS** [comment]
> The issue is [https://github.com/apache/dolphinscheduler/issues/13146](url). And the reason is when modifying the workflow definition, if I do not know the taskcode, I will not be able to modify the workflow definition. Therefore, can DS support calling the api through projectCode and processCode to obtain the corresponding taskCode, so that I can easily modify it.

We can use `queryProcessDefinitionByCode` to query the process definition detail by `projectCode` and `processDefinitionCode`. So I think this pr is unnessnary. Feel free to correct me if I'm wrong.

**Comment by LiJie20190102** [comment]
> > The issue is [https://github.com/apache/dolphinscheduler/issues/13146](url). And the reason is when modifying the workflow definition, if I do not know the taskcode, I will not be able to modify the workflow definition. Therefore, can DS support calling the api through projectCode and processCode to obtain the corresponding taskCode, so that I can easily modify it.
> 
> We can use `queryProcessDefinitionByCode` to query the process definition detail by `projectCode` and `processDefinitionCode`. So I think this pr is unnessnary. Feel free to correct me if I'm wrong.

I think queryProcessDefinitionByCode returns too much redundant information for me, which will take a long time. I just want processTaskRelation info

**Comment by LiJie20190102** [comment]
@SbloodyS Hello, can you help me merger it.

**Comment by SbloodyS** [comment]
> I think queryProcessDefinitionByCode returns too much redundant information for me, which will take a long time. I just want processTaskRelation info

I don't think this is necessary because it is an open source project, not a customized project. cc @ruanwenjun @zhongjiajie @caishunfeng @EricGao888

File: dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/service/impl/ProcessTaskRelationServiceImpl.java [hunk 1]
```diff
@@ -543,6 +543,38 @@ public Map<String, Object> deleteEdge(User loginUser, long projectCode, long pro
...
+     * @param processDefinitionCode processDefinitionCode
+     * @return  process task relation lis
+     */
+    @Override
+    public Map<String, Object> queryProcessTaskRelation(User loginUser, long projectCode,
+        long processDefinitionCode) {
+
+        Map<String, Object> result = new HashMap<>();
+
+        ProcessDefinition processDefinition = processDefinitionMapper.queryByCode(processDefinitionCode);
```
Comment (EricGao888) [inline dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/service/impl/ProcessTaskRelationServiceImpl.java:560]: DS usually checks both interface and resource permissions in service level for security purpose, see:
https://github.com/apache/dolphinscheduler/blob/7586710d49ea86a22798a3f401a875e2f8669dbc/dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/service/impl/ProcessTaskRelationServiceImpl.java#L123
Without permission check,  one user could easily hack other's data.

Comment (LiJie20190102) [reply dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/service/impl/ProcessTaskRelationServiceImpl.java:560]: > DS usually checks both interface and resource permissions in service level for security purpose, see:
> 
> https://github.com/apache/dolphinscheduler/blob/7586710d49ea86a22798a3f401a875e2f8669dbc/dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/service/impl/ProcessTaskRelationServiceImpl.java#L123
> 
> 
> Without permission check, one user could easily hack other's data.

OK, please confirm whether my requirements can be integrated. If they can be integrated, I will modify them as soon as possible

**Comment by EricGao888** [comment]
@LiJie20190102 BTW, please submit PR to `dev` branch instead of `xxx-prepare`. We always submit PR to `dev` branch for the first choice and furthermore, we usually do not accept new features into a `xxx-prepare` branch.
image

**Comment by LiJie20190102** [comment]
This is my first time to contribute code. I especially hope to contribute code to ds. Please give me a chance. I will contribute more code in the future. @SbloodyS @EricGao888 @ruanwenjun @zhongjiajie @caishunfeng

**Comment by SbloodyS** [comment]
> > I think queryProcessDefinitionByCode returns too much redundant information for me, which will take a long time. I just want processTaskRelation info
> 
> I don't think this is necessary because it is an open source project, not a customized project. cc @ruanwenjun @zhongjiajie @caishunfeng @EricGao888

I'm +0 on this. Not -1. So if others think it's OK, please continue.

**Comment by LiJie20190102** [comment]
> > > I think queryProcessDefinitionByCode returns too much redundant information for me, which will take a long time. I just want processTaskRelation info
> > 
> > 
> > I don't think this is necessary because it is an open source project, not a customized project. cc @ruanwenjun @zhongjiajie @caishunfeng @EricGao888
> 
> I'm +0 on this. Not -1. So if others think it's OK, please continue.
If others think it is OK, I will submit the code to the dev branch as soon as possible. @ruanwenjun @zhongjiajie @caishunfeng @EricGao888

**Comment by LiJie20190102** [comment]
@ruanwenjun @zhongjiajie @caishunfeng @EricGao888
Hello, can I submit my improve to the dev branch? If not, I will close the PR

**Pull request closed without merging**